### PR TITLE
Use njit to speed up connected_four func, remove BoardPiece dependency

### PR DESCRIPTION
**Pull request by iamzoltan**
Using njit from numba to speed up connected four checks. Works super well. Changed bools board check to work with the new board representation. Need to rework benchmarking branch as well. Lets get through this code quick to merge it into master and rebase our current tests @goodboy @owenmackwood @ikhwankhalid

### CODE REVIEW

File: connect_four/agents/common.py [hunk 1]
```diff
@@ -221,44 +229,47 @@ def connected_four(
...
+    else:
+        pidx = 2
+
+    for i in range(rows):
+        for j in range(col_edge):
             # check horizontal
-            hori = board[i, j:j+num_to_win, :]
-            if (hori == player).all():
+            hori = board[i, j:j+num_to_win, pidx]
+            if hori.all() == True:
```
Comment (ikhwankhalid) [inline connect_four/agents/common.py:246]: Does this do away with one-hot encoding?

Comment (iamzoltan) [reply connect_four/agents/common.py:246]: no, I decided to idx the array at the position that represents the player. Njit was complaining about the old implementation. This is still a boolean comparison

File: connect_four/main.py [hunk 2]
```diff
@@ -98,4 +96,4 @@ def human_vs_agent(
 
 if __name__ == "__main__":
     # human_vs_agent(user_move)
-    human_vs_agent(genmove_montecarlo, genmove_minimax)
+    human_vs_agent(genmove_montecarlo, genmove_minimax, init_1=connected_four)
```
Comment (ikhwankhalid) [inline connect_four/main.py:99]: What does `init_1=connected_four` do?

**Comment by iamzoltan** [comment]
do you think this is ok to merge?

**Comment by iamzoltan** [comment]
sure, whats up?

File: connect_four/main.py (hunk 2 above)
Comment (iamzoltan) [reply connect_four/main.py:99]: this passes in the connected_four functions to be called before the game starts. What this does is give njit a chance to compile the function beforehand for later use. If we did not pass it in, the we would have to wait for its first call for njit to compile and save it for later use.

Comment (ikhwankhalid) [reply connect_four/main.py:99]: Ah, okay. Yeah looks good to merge.

**Pull request merged**